llm/model/qwen3_14b_executor.py

```
@@ -0,0 +1,593 @@
+        )
+
+    @staticmethod
+    def _stack_decode_weights(layers: list[_KernelLayerWeights]) -> dict[str, torch.Tensor]:
+        # Stack from already-prepared per-layer kernel weights. Each
+        # _KernelLayerWeights field is already in the kernel-ready shape/dtype
+        # (transposed bf16 cpu for projections, [1, N] float cpu for norms),
+        # so a plain cat along dim 0 is all that's left. Reading from the
+        # original model.layers here would crash because _release_layer_weights
+        # has already replaced those tensors with torch.empty(0).
+        def cat(attr: str) -> torch.Tensor:
+            return torch.cat([getattr(l, attr) for l in layers], dim=0)
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)